backend/typescript/services/implementations/EntityServiceMg.ts

```diff
@@ -72,14 +92,41 @@ class EntityService implements IEntityService {
     entity: EntityRequestDTO,
   ): Promise<EntityResponseDTO | null> {
     let updatedEntity: Entity | null;
+    let fileName = "";
     try {
-      updatedEntity = await MgEntity.findByIdAndUpdate(id, entity, {
-        new: true,
-        runValidators: true,
-      });
+      const currentEntity = await MgEntity.findById(id, "fileName");
+      const currentFileName = currentEntity?.fileName;
+      if (entity.filePath) {
+        fileName = currentFileName || uuidv4();
+      }
+      updatedEntity = await MgEntity.findByIdAndUpdate(
+        id,
+        { ...entity, fileName },
+        {
+          new: true,
+          runValidators: true,
+        },
+      );
       if (!updatedEntity) {
         throw new Error(`Entity id ${id} not found`);
       }
+      if (entity.filePath) {
+        if (currentFileName) {
+          this.storageService.updateFile(
+            fileName,
+            entity.filePath,
+            entity.fileContentType,
+          );
+        } else {
+          this.storageService.createFile(
+            fileName,
+            entity.filePath,
+            entity.fileContentType,
+          );
+        }
+      } else if (currentFileName) {
+        this.storageService.deleteFile(currentFileName);
+      }
```

Comment:
Let's do file update/creation before entity update. Potential dangling file issue if we hit the `createFile` case: creation could fail and there would be a reference to a file that does not exist in the DB

Same thing for the PG service